index/full.go

```diff
@@ -0,0 +1,39 @@
+package index
+
+import (
+	"errors"
+
+	"github.com/ipfs/go-cid"
+)
+
+// ErrNotFound is returned when an entry is not found in an index.
+var ErrNotFound = errors.New("entry not found in index")
```

Comment:
Suggest we create a custom error struct:
```
type ErrNotFound struct {
    Message string
    c cid.Cid
}
```

Reply:
How about we use error wrapping:

```
return fmt.Errorf("entry %s: %w", cid, ErrNotFound).
```

https://blog.golang.org/go1.13-errors